cmd/snap/cmd_quota_test.go

```diff
@@ -128,11 +136,21 @@ func (s *quotaSuite) TestGetQuotaGroupSimple(c *check.C) {
 }
 
 func (s *validateSuite) TestCreateQuotaGroup(c *check.C) {
-	s.RedirectClientToTestServer(makeFakeQuotaPostHandler(c, `{"type": "sync", "status-code": 200, "result": []}`, "foo", "bar", []string{"snap-a"}, 999))
+	s.RedirectClientToTestServer(makeFakeQuotaPostHandler(c, "ensure", `{"type": "sync", "status-code": 200, "result": []}`, "foo", "bar", []string{"snap-a"}, 999))
 
 	rest, err := main.Parser(main.Client()).ParseArgs([]string{"quota", "foo", "--max-memory=999B", "--parent=bar", "snap-a"})
 	c.Assert(err, check.IsNil)
 	c.Check(rest, check.HasLen, 0)
 	c.Check(s.Stderr(), check.Equals, "")
 	c.Check(s.Stdout(), check.Equals, "")
 }
+
+func (s *validateSuite) TestRemoveQuotaGroup(c *check.C) {
```

Comment:
(nitpick^3) we could test if errors from the client are handled correctly but it seems we rarely do this in cmd_*_test.go so I think this is fine (i.e. if we mock a error from the test server)

Reply:
There is TestRemoveQuotaGroupError below, or do you mean some other case?

Reply:
Indeed, thanks for the API level tests! 

It's fine as it is, let me just explain what I had in mind but let's not do this now but finish the rest first :)

We don't check api errors at this level, e.g.:
```
diff --git a/cmd/snap/cmd_quota.go b/cmd/snap/cmd_quota.go
index 00491dfa5f..a84417e5c1 100644
--- a/cmd/snap/cmd_quota.go
+++ b/cmd/snap/cmd_quota.go
@@ -128,5 +128,6 @@ type cmdRemoveQuota struct {
 }
 
 func (x *cmdRemoveQuota) Execute(args []string) (err error) {
-       return x.client.RemoveQuotaGroup(x.Positional.GroupName)
+       x.client.RemoveQuotaGroup(x.Positional.GroupName)
+       return nil
 }
```
will not fail. So something like:
```
func (s *validateSuite) TestRemoveQuotaGroupUnhappy(c *check.C) {
	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
		w.WriteHeader(500)
		fmt.Fprintln(w, "boom")
	})

	rest, err := main.Parser(main.Client()).ParseArgs([]string{"remove-quota", "foo"})
	c.Assert(err, check.ErrorMatches, "error: cannot remove quota group: ...")
	c.Check(rest, check.HasLen, 0)
	c.Check(s.Stderr(), check.Equals, "")
	c.Check(s.Stdout(), check.Equals, "")
}
`` `
*but* this is a big discration right now because AFAICT we almost never in in the other cmd_*_test.go test this so let's do it (if at all) after we landed this feature.
```

Reply:
Ah, got you, yeah, a material for a followup, thanks!